src/main/kotlin/com/sksamuel/avro4k/schema/SchemaFor.kt

```diff
@@ -130,7 +130,9 @@ class NullableSchemaFor(private val schemaFor: SchemaFor, private val annotation
 fun schemaFor(serializersModule: SerializersModule,
               descriptor: SerialDescriptor,
               annos: List<Annotation>,
-              namingStrategy: NamingStrategy): SchemaFor {
+              namingStrategy: NamingStrategy,
+              resolvedSchemas: MutableMap<String, Schema> = mutableMapOf()
```

Comment:
Please remove the default value so that the caller needs to think about what to pass here. Doing so will reveal that the resolvedSchemas map needs to be passed to the ListSchemaFor, MapschemaFor and PairSchemaFor classes (see coment on test cases).

Reply:
nice trick